### PR TITLE
W-13735979-Review-Debug-a-Mule-app-CP

### PR DESCRIPTION
**Pull request by Crispy-Salesforce**

### CODE REVIEW

File: modules/ROOT/pages/debug-a-mule-application.adoc [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
+// TODO: The "Call Stack" panel shows an icon next to the "Running" word that is not present in the desktop IDE.
 
-== Getting to Know the Debugger
-
-image::debugger-view.png[]
-
-. *Start Debugging*: A shortcut to start the build and run your Mule application in debug mode.
-. *Variables Panel*: Shows information about the Mule event and variables.
-. *Watch Panel*: Use this panel to evaluate variables and expressions.
+. *Start Debugging*: Starts the build and run your Mule application in debug mode.
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-a-mule-application.adoc]:
```suggestion
. *Start Debugging*: Starts the build and runs your Mule application in debug mode.
```

File: modules/ROOT/pages/debug-a-mule-application.adoc [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
+. *Breakpoints Panel*: Summarizes all the breakpoints in your Mule application.
+. *Terminal*: Outputs the Maven build and the results of the deployment to the embedded Mule Runtime engine.
+. *Debug Toolbar*: Enables you to navigate through your running Mule application.
 +
-image::debugger-toolbar.png[]
+image::debugger-toolbar.png["Debugger toolbar buttons highlighted"]
 [arabic]
-.. *Step Over*: If the line does not contain a `flowref` it behaves the same as next, it takes `flowref` as a unit of process avoiding to stop until return to the current flow.
-.. *Step Into*: If the line does not contain a `flowref` it behaves the same as next but if it does the debugger enters the flow referenced.
+.. *Step Over*: If the line does not contain a `flowref`, it behaves the same as next. It takes `flowref` as a unit of process avoiding to stop until return to the current flow.
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-a-mule-application.adoc]: next line? option?

Comment (hannanelson) [reply modules/ROOT/pages/debug-a-mule-application.adoc]: I don't understand this:
It takes `flowref` as a unit of process avoiding to stop until return to the current flow.

File: modules/ROOT/pages/debug-a-mule-application.adoc [hunk 3]
```diff
@@ -1,33 +1,35 @@
...
+.. *Step Into*: If the line does not contain a `flowref`, it behaves the same as next. If the line contains a `flowref`, the debugger enters the flow referenced.
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-a-mule-application.adoc]: next what?

File: modules/ROOT/pages/debug-a-mule-application.adoc [hunk 4]
```diff
@@ -1,33 +1,35 @@
...
 .. *Step Out*: The debugger goes out of the current flow trying to return to a previous flow. If another flow doesn’t exist, the debugger behaves like *continue*.
+// TODO: Am I missing something? Where is *continue* explained?
 .. *Restart*: Not supported.
+// TODO: Can we request assistance from technical team to learn why this feature is not supported now? Will it be supported later?
 
-The following section teaches you how to set up components that help you debug your Mule application, set up breakpoints for your code, and run a debug session.
+In the following section you learn how to set up components that help you debug your Mule application, set up breakpoints for your code, and run a debug session.
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-a-mule-application.adoc]:
```suggestion
The following sections explain how to set up components that help debug your Mule application, set up breakpoints for your code, and run a debug session.
```

File: modules/ROOT/pages/debug-add-breakpoints.adoc [hunk 5]
```diff
@@ -22,21 +23,20 @@ To add a breakpoint to your Mule application:
...
-image::add-breakpoint.png[]
+image::add-breakpoint.png["Breakpoint highlighted next to the line number"]
 +
-[NOTE]
---
-You can also use `F9` to add a breakpoint to the line you are standing on.
---
-. On your `implementation.xml` file, add two more breakpoints for the `<db:select>` and the `<set-variable>` components.
-. Note that all your breakpoints show up in the *Breakpoints* panel:
+NOTE: You can also use `F9` to add a breakpoint to the line you are standing on.
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-add-breakpoints.adoc]:
```suggestion
NOTE: You can also use `F9` to add a breakpoint to the current line.
```

File: modules/ROOT/pages/debug-add-breakpoints.adoc [hunk 6]
```diff
@@ -22,21 +23,20 @@ To add a breakpoint to your Mule application:
...
 +
-image::breakpoints-panel.png[]
+. On your `implementation.xml` file, add two more breakpoints for the `<db:select>` and the `<set-variable>` components. Your breakpoints show up in the *Breakpoints* panel:
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-add-breakpoints.adoc]:
```suggestion
. In your `implementation.xml` file, add two more breakpoints for the `<db:select>` and the `<set-variable>` components. Your breakpoints appear in the *Breakpoints* panel:
```

File: modules/ROOT/pages/debug-add-breakpoints.adoc [hunk 7]
```diff
@@ -22,21 +23,20 @@ To add a breakpoint to your Mule application:
...
+// TODO: Panel name is always in capital letters both in web IDE and in desktop IDE.
+
+image::breakpoints-panel.png["Breakpoints added highlighted"]
 
 == Next Step
 
 * xref:run-a-debug-session.adoc[Run a Debug Session]. +
-Run your first debug session and inspect the message and variables at different points during your application's execution.
+Run your first debug session and inspect the message and variables at different points during your application execution.
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-add-breakpoints.adoc]:
```suggestion
Run your first debug session and inspect the message and variables at different points during application execution.
```

File: modules/ROOT/pages/debug-add-logger-set-variables.adoc [hunk 8]
```diff
@@ -9,17 +9,20 @@ You can use Logger and Set Variable components to help debug your Mule applicati
 Before you begin, ensure that:
 
 * You designed an API specification. +
```
Comment (hannanelson) [inline modules/ROOT/pages/debug-add-logger-set-variables.adoc]: We need to change all these + to a line by itself.

File: modules/ROOT/pages/run-a-debug-session.adoc [hunk 9]
```diff
@@ -51,13 +51,11 @@ image::debugger-stop-set-variable.png[]
 +
 image::debug-add-expression.png[]
 . Add the expression `payload.^mediaType`.
-. Select the *Step Over* button until your execution moves to the `<set-payload>` component.
-. Note that your payload is `application/java`.
-. Select the *Step Over* button until your execution jumps to the `<logger>` component.
-. Note that your payload is now transformed to `application/json`.
+. Select the *Step Over* button until your execution moves to the `<set-payload>` component. Your payload is `application/java`.
+. Select the *Step Over* button until your execution jumps to the `<logger>` component. Your payload is now transformed to `application/json`.
```
Comment (hannanelson) [inline modules/ROOT/pages/run-a-debug-session.adoc]: In previous step, jumps is moves.

Comment (hannanelson) [reply modules/ROOT/pages/run-a-debug-session.adoc]: Should these steps be parallel?

**Review by hannanelson — DISMISSED** [review]

File: modules/ROOT/pages/debug-add-logger-set-variables.adoc (hunk 8 above)
Comment (dukesphere) [reply modules/ROOT/pages/debug-add-logger-set-variables.adoc]: Those drive me NUTS! I feel like I've been asking people not to do that for years now. 

/vent

File: modules/ROOT/pages/debug-a-mule-application.adoc (hunk 3 above)
Comment (dukesphere) [reply modules/ROOT/pages/debug-a-mule-application.adoc]: TODO duke: Need to understand. I'm not clear either.

File: modules/ROOT/pages/debug-a-mule-application.adoc (hunk 2 above)
Comment (dukesphere) [reply modules/ROOT/pages/debug-a-mule-application.adoc]: TODO duke: Need to understand. I'm not clear either.

File: modules/ROOT/pages/debug-a-mule-application.adoc (hunk 3 above)
Comment (dukesphere) [reply modules/ROOT/pages/debug-a-mule-application.adoc]: working on this; got info from guillermo

**Pull request merged**